src/components/BookForm.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
+
+const BookForm = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('');
+
+  const addNewBook = () => {
+    const newBook = {
+      title,
+      category,
+      id: uuidv4(),
+    };
+
+    dispatch(addBook(newBook));
+    setTitle('');
+    setCategory('');
+  };
+  const handelTitleChange = (e) => {
+    setTitle(e.target.value);
+  };
+  const handelCategoryChange = (e) => {
+    setCategory(e.target.value);
+  };
+  return (
+    <div>
+      <h5>ADD NEW BOOK</h5>
+      <input
+        placeholder="Book Title"
+        type="text"
+        onChange={handelTitleChange}
+        value={title}
+        required
+      />
+      <select
+        id="cars"
+        name="cars"
+        onChange={handelCategoryChange}
+        value={category}
+      >
+        <option>Action</option>
+        <option>Comedy</option>
+        <option>Drama</option>
+        <option>Horror</option>
+        <option>Romance</option>
+      </select>
+      <button type="button" value="ADD BOOK" onClick={addNewBook}>
+        ADD NEW BOOK
+      </button>
```

Comment:
[OPTIONAL] I think it could be nice if you either wrap the inputs around a form tag and change the button to `type="submit"` or add `onKeyPress` attribute to the button and filter for the enter key. This to variations would enable books to be submitted by pressing the enter key. This would be good for the accessibility of the app.

Also adding labels to the inputs could be a nice improvement.